### PR TITLE
You can only attach to running containers

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Trying to remove some TODO statements from code
@baude @mheon @umohnani8 PTAL

**Comment by mheon** [comment]
This breaks kpod run. We need to attach to created containers (attach fires before start to fix a race condition).

**Comment by rhatdan** [comment]
So it should only attach to a configured or running container.

**Comment by mheon** [comment]
I added stopped to that because, when we add the ability to restart containers, we'll need to attach to the stopped container before firing start.

File: libpod/container.go [hunk 1]
```diff
@@ -598,11 +598,9 @@ func (c *Container) Attach(noStdin bool, keys string, attached chan<- bool) erro
 		return err
 	}
 
-	// TODO is it valid to attach to a frozen container?
-	if c.state.State == ContainerStateUnknown ||
-		c.state.State == ContainerStateConfigured ||
-		c.state.State == ContainerStatePaused {
-		return errors.Wrapf(ErrCtrStateInvalid, "can only attach to created, running, or stopped containers")
+	if c.state.State != ContainerStateConfigured &&
```
Comment (mheon) [inline libpod/container.go]: Created, not configured - Configured is not in runc yet

**Comment by mheon** [comment]
Still, we're probably a ways off getting restarting working, so I'm fine with keeping things this way until we do. One more thing then LGTM

**Comment by rhatdan** [comment]
Tests are passing.
PTAL @TomSweeneyRedHat @baude @umohnani8 And merge.

**Comment by umohnani8** [comment]
LGTM

**Comment by mheon** [comment]
LGTM
@rh-atomic-bot r+

**Pull request closed without merging**